src/Tools/dotnet-monitor/ServiceCollectionExtensions.cs

```diff
@@ -202,22 +208,23 @@ public static IServiceCollection ConfigureDiagnosticPort(this IServiceCollection
             return services;
         }
 
-        private static IServiceCollection RegisterProvider<TOptions, TProvider>(this IServiceCollection services, string name)
+        private static IServiceCollection RegisterEgressType<TOptions, TProvider>(this IServiceCollection services)
             where TProvider : class, IEgressProvider<TOptions>
-            where TOptions : class
+            where TOptions : class, new()
         {
+            // These Singletons are "IEnumerable<T>" services where there are multiple services registered (if TOptions is the same)
             // Add services to provide raw configuration for the options type
-            services.AddSingleton(sp => new EgressProviderConfigurationProvider<TOptions>(sp.GetRequiredService<IConfiguration>(), name));
+            services.AddSingleton<EgressProviderConfigurationProvider<TOptions>>();
             services.AddSingletonForwarder<IEgressProviderConfigurationProvider<TOptions>, EgressProviderConfigurationProvider<TOptions>>();
             services.AddSingletonForwarder<IEgressProviderConfigurationProvider, EgressProviderConfigurationProvider<TOptions>>();
 
+            services.AddSingleton<IOptionsChangeTokenSource<TOptions>, EgressProviderConfigurationChangeTokenSource<TOptions>>();
```

Comment:
Can you move this back to where it was so that all of the `IOptionsChangeTokenSource<TOptions>` implementations are grouped together?